src/main/java/mSearch/daten/ListeFilme.java

```diff
@@ -336,24 +258,13 @@ public String getFileSizeUrl(String url, String sender) {
      * @param sender Sender which films are to be deleted.
      */
     public synchronized void deleteAllFilms(String sender) {
-        DatenFilm film;
-        Iterator<DatenFilm> it = this.iterator();
-        while (it.hasNext()) {
-            film = it.next();
-            if (film.arr[DatenFilm.FILM_SENDER].equalsIgnoreCase(sender)) {
-                it.remove();
-            }
-        }
+        removeIf(film -> film.arr[DatenFilm.FILM_SENDER].equalsIgnoreCase(sender));
     }
 
 
     public synchronized DatenFilm getFilmByUrl(final String url) {
-        for (DatenFilm film : this) {
-            if (film.arr[DatenFilm.FILM_URL].equalsIgnoreCase(url)) {
-                return film;
-            }
-        }
-        return null;
+        Optional<DatenFilm> opt = this.parallelStream().filter(f -> f.arr[DatenFilm.FILM_URL].equalsIgnoreCase(url)).findAny();
+        return opt.orElse(null);
     }
 
     public synchronized void checkThema(String sender, LinkedList<String> liste, String thema) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') The type of the "liste" object should be an interface such as "List" rather than the implementation "LinkedList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1319)